ThirdPartyAdapters/chartboost/chartboost/src/main/java/com/google/ads/mediation/chartboost/ChartboostSingleton.java

```diff
@@ -128,33 +145,47 @@ private static void addRewardedDelegate(String location,
         return null;
     }
 
+    protected static WeakReference<ChartboostBanner> getChartboostBanner(String location) {
+        if (!TextUtils.isEmpty(location) && mBannerMap.containsKey(location)) {
+            return mBannerMap.get(location);
+        }
+        return null;
+    }
+
+    protected static WeakReference<ChartboostBanner> detachBanners() {
+        Collection<WeakReference<ChartboostBanner>> banners = mBannerMap.values();
+        for(WeakReference<ChartboostBanner> ref : banners) {
+            if(ref != null && ref.get()!=null) {
+                ref.get().detachBanner();
+                ref.clear();
+            }
+        }
+        return null;
+    }
+
     /**
      * This method will initialize Chartboost SDK for interstitial ads and return whether or not
      * it successfully initialized.
      *
      * @param context         Required to initialize Chartboost SDK.
      * @param adapterDelegate The adapter delegate to which to forward Chartboost callbacks.
      */
-    public static void startChartboostInterstitial(
-            Context context, AbstractChartboostAdapterDelegate adapterDelegate) {
+    public static void startChartboostInterstitial(Context context, AbstractChartboostAdapterDelegate adapterDelegate) {
         String location = adapterDelegate.getChartboostParams().getLocation();
-
         // Checks if an ad has already been sent for caching for the requested location, and fail
         // the ad request if it is.
-        WeakReference<AbstractChartboostAdapterDelegate> delegate =
-                getInterstitialDelegate(location);
+        WeakReference<AbstractChartboostAdapterDelegate> delegate = getInterstitialDelegate(location);
         if (delegate != null && delegate.get() != null) {
-            Log.w(ChartboostMediationAdapter.TAG,
-                    "An ad has already been requested for the location: " + location);
-            adapterDelegate.didFailToLoadInterstitial(location,
-                    CBError.CBImpressionError.NO_AD_FOUND);
+            Log.w(ChartboostMediationAdapter.TAG, "An ad has already been requested for the location: " + location);
+            adapterDelegate.didFailToLoadInterstitial(location, CBError.CBImpressionError.NO_AD_FOUND);
+            mInterstitialDelegates.remove(location);
```

Comment:
Not really sure about this. Only applies to the same location but if you don't remove the references all the ads are not displayed.

Reply:
Which ads are not displayed? The current ad request should fail since we are returning in the next line. But it seems to me that by doing `mInterstitialDelegates.remove(location);` you are preventing the previous ad from being displayed.

Reply:
Fixed